Add ActionsToolbar component

dsas opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- Link this PR to its associated issue with an appropriate keyword: Closes, See, Follow up to, etc. -->
Closes <!-- #ISSUE-NUMBER or URL -->

<!-- In a few words, what is the PR actually doing? -->

Creates a new ActionsToolbar component that lets non-DataViews code display a list of Actions.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

So DataView consumers can re-use the actions they've defined for use within DataViews with eligibility rules, callbacks etc, in other parts of their UI.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

It creates a new `ActionsToolbar` component, which understands Actions. 

Consumers supply an item and a list of actions, the ActionsToolbar then checks which actions are eligible for use with this Item and displays a list of those that are. 

Depending upon the props given, it's essentially the same as the internal `PrimaryActions`, `PostActions` and `CompactItemActions` components worked before - in fact this component replaces those components and they're removed as part of this PR.


## Testing Instructions

A storybook is provided under DataViews → ActionsToolbar

For a real-life use:
1. Open up the site editor
2. Go to Pages
3. Checkout the actions in the list - the icons should display and should work as before
4. Switch to Table View to see a longer list of actions - again the icons should display and work as before.
5. Select a page
6. Press the sidebar icon next to the dataview settings
7. The quick edit panel should appear
8. Press the three dot 'more' menu - this time the actions should display and work as before -- as a menu

<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

1. Follow the steps as above
2. You should be able to tab through the actions
3. You should be able to hit enter on the actions to activate them

## Screenshots or screencast <!-- if applicable -->

<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

The screenshots should look identical

|Before|After|
|-|-|
<img width="1423" height="548" alt="Screenshot 2025-07-23 at 23 01 20" src="https://github.com/user-attachments/assets/cb974e30-9314-49cd-a755-0f401469442a" /> | <img width="1423" height="548" alt="Screenshot 2025-07-23 at 23 02 12" src="https://github.com/user-attachments/assets/b05213e6-8124-40bc-b0e0-59f7574aae14" />

dsas committed:
DataViews: Add ActionsToolbar component

This is a toolbar which displays DataView actions as a row of buttons.
As appropriate for the given items. It can be used outside of the
DataViews UI itself.

dsas committed:
Add showLabels to use text rather than icons

dsas committed:
Replace PrimaryActions with ActionsToolbar

ActionsToolbar can do everything PrimaryActions can do, it was most of
the inspiration behind it.

youknowriad commented on packages/dataviews/src/index.ts:
```diff
@@ -1,6 +1,8 @@
 export { default as DataViews } from './components/dataviews';
 export { default as DataForm } from './components/dataform';
+export { default as ActionsToolbar } from './components/actions-toolbar';
```
Can we try to use it in the quick edit panel of the site editor for instance, I think that will clarify the need for this component better and ensure we have the right API... for it.

For instance, I think we might need a prop that says: render all actions in the dropdown including primary actions.

dsas replied:
I added a `variant` prop to choose between Menu and Toolbar and also a `size` (small, compact, undefined) prop which is passed on to the underlying Menu.

Adding the size prop also let me drop the CompactItemActions component used in dataviews.

Kinda getting a little bit antsy at replacing too many things in one go, but maybe  the Action HStacks / Menus in `dataviews-layouts/list/index.tsx` could be replaced too, though it's a little more complicated perhaps.

dsas committed:
Add menu variant

dsas committed:
Replace CompactItemActions with ActionsToolbar

dsas committed:
Replace PostActions with ActionsToolbar

dsas committed:
Remove PostActions component - all uses have now been replaced by ActionsToolbar

dsas commented on packages/editor/src/components/post-actions/index.js:
```diff
@@ -1,147 +0,0 @@
-}
-
-// From now on all the functions on this file are copied as from the dataviews packages,
-// The editor packages should not be using the dataviews packages directly,
```
I'm a little bit concerned about this - I've broken this rule by using ActionsToolbar in `editor/src/components/post-card-panel/index.js`

youknowriad replied:
This is likely to cause the dataviews package to be bundled twice, once within the editor package and once within edit-site. I'm not sure yet what the impact is.

oandregal replied:
We'd bundle it twice if we imported from `wordpress/dataviews/wp`, right? I see that we do `import { ActionsToolbar } from '@wordpress/dataviews';` in post-card-panel, so it seems we're fine here.

I understand why we'd want to prevent using `editor` from `dataviews`: editor is higher-level. I don't know why we wouldn't want editor to use dataviews (other than for the bundling thing). I've checked the [original PR](https://github.com/WordPress/gutenberg/pull/60486) that introduced this (and [previous work](https://github.com/WordPress/gutenberg/pull/59849)) and I don't see the rationale. So the only thing that comes to mind if that we didn't want this component to be public? 🤔

oandregal replied:
Thinking of alternate places where this component could live, it's weird everywhere:

- components package: this is for low-level primitives, like this could be. But then, it's very related to dataviews in that it works with the higher-level Item / Actions.
- fields package: this already exports some UI components, but it's focused on WordPress-only things. This new component would be generic.

dsas replied:
> So the only thing that comes to mind if that we didn't want this component to be public?

Well, that's my main motivation - I want to be able to use the item actions I've defined both inside and outside of DataViews. I'd rather not maintain my own actions list and ActionModal UI. We have two copies of ActionModal in Gutenberg right now, how many more will proliferate around the ecosystem?

I see what you mean about it being weird everywhere though.

dsas commented on packages/dataviews/src/components/actions-toolbar/index.tsx:
```diff
@@ -0,0 +1,205 @@
+								label={ __( 'Actions' ) }
+								accessibleWhenDisabled
+								disabled={ ! eligibleActions.length }
+								className={ `dataviews-all-actions-button ${ className }` }
```
Probably shouldn't have this `dataviews-all-actions-button` classname here, I've taken it from the existing use, but it probably needs renaming / refactoring. It's not necessarily all of the actions for this item being passed here.

dsas committed:
Display the menu variant even when there are no actions

The menu variant is expected to display itself even when no actions are
available.

This fixes the e2e tests related to template reverting.

dsas committed:
Use a more neutral CSS class name

oandregal commented on packages/dataviews/src/components/actions-toolbar/index.tsx:
```diff
@@ -0,0 +1,209 @@
+ * @param props.variant    - Rendering style: 'toolbar' for horizontal buttons, 'menu' for dropdown (default: 'toolbar')
+ * @param props.size       - Button size: 'small', 'compact', or undefined for default (menu variant only)
+ */
+export default function ActionsToolbar< Item >( {
```
Why don't we export the existing `ItemActions` directly?

For example, some of the checks we have here for eligibleActions are already taken care of in `ItemActions` so we're running them twice.

dsas replied:
Could do 🤔 

A key difference is that ItemActions currently has two modes
 * isCompact - aka a menu
 * Default - Primary Actions are icon-only, other buttons are added to a menu

I think I'd like to have a mode which is essentially "All actions as proper buttons not icons", maybe that could overflow into a menu if space requires.

However, taking a step back, I can start with Default and _then_ evaluate if there's a need for my mooted "All actions as proper button" mode.
